src/frontend/src/expr/type_inference/func.rs

```diff
@@ -623,6 +623,20 @@ fn infer_type_for_special(
                 _ => Ok(None),
             }
         }
+        ExprType::TrimArray => {
+            ensure_arity!("trim_array", | inputs | == 2);
+
+            if inputs[0].is_unknown() || inputs[1].is_unknown() {
+                return Err(ErrorCode::BindError("unknown type".to_string()).into());
+            }
+
+            match (inputs[0].return_type(), inputs[1].return_type()) {
```

Comment:
The 2 parameters can be checked separately, and the latter can be `.cast_implicit(DataType::Int32)`. This means we accept `smallint` and `unknown` but reject `bigint`, as in PostgreSQL:

```
test=# SELECT trim_array(array[1, 2], 1::smallint);
 trim_array 
------------
 {1}
(1 row)

test=# SELECT trim_array(array[1, 2], 1::bigint);
ERROR:  function trim_array(integer[], bigint) does not exist
LINE 1: SELECT trim_array(array[1, 2], 1::bigint);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

test=# SELECT trim_array(array[1, 2], '1');
 trim_array 
------------
 {1}
(1 row)
```

Note the `.cast_implicit` takes ownership and you may need `std::mem::replace` with a temporary dummy value. You can find some examples in the same file.